Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +18,114 @@ public class UserCommunicateService : IUserCommunicateService
             {
-                Console.WriteLine(Constants.Validation.EmptyCityName);
-                _logger.LogInformation("The user entered an empty city name. Try again");
-                return true;
+                case 0:
+                    _invoker.SetCommand(new CurrentWeatherCommand(this));
+                    break;
+                case 1:
+                    _invoker.SetCommand(new ForecastWeatherCommand(this));
+                    break;
+                case 2:
+                    _invoker.SetCommand(new ExitCommand());
+                    break;
             }
 
-            var weatherRepresentation = (string)default;
             try
             {
-                weatherRepresentation = (await _weatherServiсe.GetByCityNameAsync(cityName)).GetStringRepresentation();
+                await _invoker.RunAsync();
             }
             catch (HttpRequestException ex)
             {
                 if (ex.StatusCode == HttpStatusCode.NotFound)
                 {
                     Console.WriteLine(Constants.Errors.BadCityName);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)HttpStatusCode.NotFound} {HttpStatusCode.NotFound}. User entered incorrect city name.");
-                    return true;
+                    _logger.LogError($"Status code: {ex.StatusCode.GetStringRepresentation()} User entered incorrect city name.");
                 }
                 else
                 {
                     Console.WriteLine(Constants.Errors.RequestError);
-                    _logger.LogError($"{DateTime.Now}| Status code: {(int)ex.StatusCode} {ex.StatusCode}. {ex.Message}");
-                    return true;
+                    _logger.LogError($"Status code: {ex.StatusCode.GetStringRepresentation()}");
+                }
+            }
+            catch (ValidationException ex)
+            {
+                foreach (var error in ex.Errors)
+                {
+                    _logger.LogError(error.ErrorMessage);
+                    Console.WriteLine(error.ErrorMessage);
                 }
             }
             catch (Exception ex)
             {
                 Console.WriteLine(Constants.Errors.UnexpectedError);
-                _logger.LogError($"{DateTime.Now}| {ex.Message}");
-                return true;
+                _logger.LogError(ex.Message);
+            }
+        }
+
+        public async Task GetCurrentWeatherAsync()
+        {
+            Console.WriteLine("Please, enter city name:");
+            var weather = await _weatherServiсe.GetByCityNameAsync(Console.ReadLine());
+
+            Console.WriteLine(weather.GetStringRepresentation());
+        }
+
+        public async Task GetForecastByCityNameAsync()
+        {
+            Console.WriteLine("Please, enter city name:");
+            string cityName = Console.ReadLine();
+
+            Console.WriteLine("Please, enter count day:");
+            int countDay;
+
+            while (true)
```

Comment:
better to make similar logic like in menu, not just infinite loop without any message to user